src/bundle/Core/Command/ExpireUserPasswordsCommand.php

```diff
@@ -0,0 +1,383 @@
+                    return $repository->getContentService()->find($filter);
+                });
+
+                /** @var \eZ\Publish\API\Repository\Values\Content\Content $content */
+                foreach ($contentList as $content) {
+                    $fields = $content->getFields();
+                    $userField = $this->findUserField($fields);
+
+                    if (null === $userField) {
+                        // content item isn't of a user content type
+                        continue;
+                    }
+
+                    $contentType = $content->getContentType();
+                    $fieldDefinition = $contentType->getFieldDefinition($userField->fieldDefIdentifier);
+
+                    $validatorConfiguration = $fieldDefinition->getValidatorConfiguration();
+                    $fieldSettings = $fieldDefinition->getFieldSettings();
+
+                    if (
+                        (
+                            !$validatorConfiguration['PasswordValueValidator']['requireNewPassword']
+                            || 0 === $fieldSettings['PasswordTTL']
+                        )
+                        && !in_array($contentType->identifier, $processedContentTypes, true)
+                    ) {
+                        $output->writeln(sprintf(
+                            '<info>Content Type "%s" needs to be updated</info>',
+                            $contentType->identifier
+                        ));
+
+                        //  enforce CT to use password expiration feature
+                        $validatorConfiguration['PasswordValueValidator']['requireNewPassword'] = true;
+                        $fieldSettings['PasswordTTL'] = (int) $passwordTTL;
+
+                        $this->updateContentType(
+                            $contentType,
+                            $fieldSettings,
+                            $validatorConfiguration,
+                            $fieldDefinition
+                        );
+
+                        $processedContentTypes[] = $contentType->identifier;
+                    }
+
+                    $spiUser = $this->userHandler->load($content->id);
+                    $updatedUser = clone $spiUser;
+                    $updatedUser->passwordUpdatedAt = 1;
+                    $this->userHandler->updatePassword($updatedUser);
+
+                    ++$processedUsersCount;
+
+                    if ($displayProgressBar) {
+                        $progressBar->advance(1);
+                    }
+                }
+            } while ($processedUsersCount < $count);
+        } catch (Exception $e) {
+            $this->repository->rollback();
+            $output->writeln(
+                '<error>Something went wrong. See the exception below, '
+                . 'fix the issue and rerun this command</error>'
+            );
+
+            throw $e;
+        }
```

Comment:
Could you please refactor this big block into smaller parts so it's more readable? Maybe at least contents of try-catch could be refactored into a method with maybe some inner methods call. Cognitive complexity now is very high.